src/main/controllers/features/sole-trader/nameController.ts

```diff
@@ -36,10 +36,12 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
             });
         } else {
             const session: Session = req.session as any as Session;
-            const acspData : ACSPData = {
-                firstName: req.body["first-name"],
-                lastName: req.body["last-name"]
-            };
+            const acspData : ACSPData = session?.getExtraData(USER_DATA)!;
+            if (acspData) {
+                acspData.firstName = req.body["first-name"];
```

Comment:
if acspData is undefined then you will have to create the object first.

Reply:
if (acspData) will always be true